packages/core/src/providers/clerk.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * <div style={{display: "flex", justifyContent: "space-between", alignItems: "center"}}>
+ * <span style={{fontSize: "1.35rem" }}>
+ *  Built-in sign in with <b>Clerk</b> integration.
+ * </span>
+ * <a href="https://clerk.com" style={{backgroundColor: "black", padding: "12px", borderRadius: "100%" }}>
+ *   <img style={{display: "block"}} src="https://authjs.dev/img/providers/clerk.svg" width="24"/>
+ * </a>
+ * </div>
+ *
+ * @module providers/clerk
+ */
+import type { OAuthConfig, OAuthUserConfig } from "./index.js"
+
+/** @see [Get the authenticated user](https://clerk.com/docs/reference/frontend-api/tag/OAuth2-Identify-Provider#operation/getOAuthUserInfo). */
+export interface ClerkProfile {
+  object: "string"
+  instance_id: "string"
+  email: "string"
+  email_verified: boolean
+  family_name: "string"
+  given_name: "string"
+  name: "string"
+  username: "string"
+  picture: "string"
+  user_id: "string"
+  public_metadata?: any
+  private_metadata?: any
+  unsafe_metadata?: any
+}
+
+/**
+ * Add Clerk login to your page.
+ *
+ * ### Setup
+ *
+ * #### Callback URL
+ * ```
+ * https://example.com/api/auth/callback/clerk
+ * ```
+ *
+ * #### Configuration
+ * ```ts
+ * import { Auth } from "@auth/core"
+ * import Clerk from "@auth/core/providers/clerk"
+ *
+ * const request = new Request(origin)
+ * const response = await Auth(request, {
+ *   providers: [
+ *     Clerk({
+ *       clientId: CLERK_CLIENT_ID,
+ *       clientSecret: CLERK_CLIENT_SECRET,
+ *       baseUrl: CLERK_BASE_URL
+ *     }),
```

Comment:
Should be 
```suggestion
 *     Clerk,
```

This should be enough, if the user sets `AUTH_CLERK_ID`, `AUTH_CLERK_SECRET` and `AUTH_CLERK_ISSUER`